metadata-ingestion/src/datahub/ingestion/source/sql/mssql/job_models.py

```diff
@@ -11,18 +15,65 @@
 
 @dataclass
 class ProcedureDependency:
+    flow_id: str
+    env: str
+    server: Optional[str]
     db: str
     schema: str
     name: str
     type: str
-    env: str
-    server: Optional[str]
+    incoming: Union[str, int]
+    outgoing: Union[str, int]
     source: str = "mssql"
 
+    def __post_init__(self):
+        self.incoming = int(self.incoming)
+        self.outgoing = int(self.outgoing)
+
 
 @dataclass
 class ProcedureLineageStream:
-    dependencies: List[ProcedureDependency]
+    dependencies: List[ProcedureDependency] = field(default_factory=list)
+
+    @property
+    def as_input_datasets(self) -> List[str]:
+        return [
+            make_dataset_urn_with_platform_instance(
+                platform=dep.source,
+                name=".".join([dep.db, dep.schema, dep.name]),
+                platform_instance=dep.server,
+                env=dep.env,
+            )
+            for dep in self.dependencies
+            if dep.type in ("U ", "V ") and dep.incoming
+        ]
+
+    @property
+    def as_output_datasets(self) -> List[str]:
+        return [
+            make_dataset_urn_with_platform_instance(
+                platform=dep.source,
+                name=".".join([dep.db, dep.schema, dep.name]),
+                platform_instance=dep.server,
+                env=dep.env,
+            )
+            for dep in self.dependencies
+            if dep.type in ("U ", "V ") and dep.outgoing
+        ]
+
+    @property
+    def as_input_datajobs(self) -> List[str]:
```

Comment:
same for the rest of these methods

```suggestion
    def get_input_datajobs(self) -> List[str]:
```

Reply:
Renamed for all methods with same naming pattern.